OpenRA.Game/Network/ReplayConnection.cs

```diff
@@ -81,6 +93,45 @@ public ReplayConnection(string replayFilename)
 						TickCount = Math.Max(TickCount, frame);
 					}
 				}
+
+				// Get last frame for all clients
+				{
+					foreach (var clientId in clientIds)
+					{
+						int lastFrame = clientsFrame[clientId]
+							.Select(x => x)
+							.Where(x => x != int.MaxValue)
+							.OrderBy(x => x).LastOrDefault();
+						lastClientsFrame[clientId] = lastFrame;
+					}
+				}
+
+				Queue<Chunk> tmpChunks = new Queue<Chunk>();
+
+				// 2nd parse : replace all disconnect packets without frame with real
+				// disconnect frame
+				// NOTE: to modify/remove if a reconnect feature is set
+				foreach (var tmpChunk in chunks)
+				{
+					foreach (var tmpPacketPair in tmpChunk.Packets)
+					{
+						var client = tmpPacketPair.First;
+						var packet = tmpPacketPair.Second;
+						if (packet.Length == 5 && packet[4] == 0xBF)
+						{
+							int lastClientFrame = lastClientsFrame[client];
+							byte[] lastFramePacket = BitConverter.GetBytes(lastClientFrame);
+							if (lastFramePacket.Length == 4)
+							{
+								Array.Copy(lastFramePacket, packet, lastFramePacket.Length);
+							}
+						}
+					}
+
+					tmpChunks.Enqueue(tmpChunk);
+				}
+
+				chunks = tmpChunks;
```

Comment:
It doesn't look like you need the temporary ```tmpChunks``` queue. It appears like you could just update the ```packet``` in place and everything would be fine. Please let me know if I've missed something that means I'm wrong.